src/events/message/messageUpdate.ts

```diff
@@ -19,6 +19,10 @@ export default class extends Event {
 	}
 
 	public async run(oldMessage: Message, newMessage: Message): Promise<void> {
+		if (newMessage.partial) {
```

Comment:
Do you happen know if the oldMessage can also be a partial or just the newMessage? A bit unsure myself

Reply:
If we assume that oldMessage can be a partial, and we take this same strategy, we're going to end up with two copies of the same message, because when we try to fetch the oldMessage, we will get the newMessage because the message was edited and the oldMessage doesn't exist anymore. All we really need from the oldMessage is information about content anyway, which hopefully at least some of is included in a partial otherwise it's not a very useful event. We can assume the author doesn't change between edits, so having just the newMessage fully hydrated is sufficient in that respect.